src/tools/perf/perftest.c

```diff
@@ -820,44 +812,39 @@ static void ctx_free(struct perftest_context *ctx)
     free(ctx->params.super.msg_size_list);
 }
 
-static perftest_rte_t plugin_sock_rte = {
-    .type = PERFTEST_RTE_SOCK,
-    .ops  = {
-        .setup   = setup_sock_rte,
-        .cleanup = cleanup_sock_rte
-    }
-};
-
-perftest_rte_t *rte_list = NULL;
+UCS_LIST_HEAD(rte_list);
 
-static const perftest_rte_t *rte;
+static ucx_perf_rte_t *rte;
 
 static void initialize_plugins(void)
 {
-    perftest_rte_list_add(&plugin_sock_rte);
 #if defined (HAVE_MPI)
-    perftest_rte_list_add(&plugin_mpi_rte);
+    ucs_list_add_tail(&rte_list, &mpi_rte.list);
 #endif
+    ucs_list_add_tail(&rte_list, &sock_rte.list);
 }
 
 static ucs_status_t setup_rte(struct perftest_context *ctx)
 {
     ucs_status_t status;
 
-    for (rte = rte_list; rte; rte = rte->next) {
-        status = rte->ops.setup(ctx);
+    ucs_list_for_each(rte, &rte_list, list) {
+        status = rte->setup(ctx);
         if (status != UCS_ERR_UNSUPPORTED) {
+            ctx->params.super.report_func = rte_report;
+            ctx->params.super.report_arg  = ctx;
             return status;
         }
     }
 
     ucs_error("No supported RTE found");
+    rte = NULL;
```

Comment:
can we remove the global var rte, it's saved on perftest context anyway

Reply:
fixed